cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/CaReconciler.java

```diff
@@ -352,32 +359,38 @@ Future<Void> clusterOperatorSecret(Supplier<Date> dateSupplier) {
      * due to a new CA key. It is not necessary when the CA certificate is replace while retaining the existing key.
      */
     Future<Void> rollingUpdateForNewCaKey() {
```

Comment:
I don't really like how you changed this method. This method used the same approach for rolling both Zoo and Kafka. But your change cause the rolling for each of them to divert. The way you do it by just doing things twice - each time in a different way - seems very hacky. Maybe a better way would be to completely split it into separate methods?

Reply:
Okay, I've refactored that.